src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/__init__.py

```diff
@@ -5,105 +5,186 @@
+    from azure.cli.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.resource.resources import ResourceManagementClient
+    from azure.cli._profile import Profile
+    # Determine if user logged in
+    try:
+        list(get_mgmt_service_client(ResourceManagementClient).resources.list())
+    except CLIError:
+        # Not logged in
+        import getpass
+        login_successful = False
+        while not login_successful:
+            method_index = prompt_choice_list(MSG_PROMPT_LOGIN, LOGIN_METHOD_LIST)
+            profile = Profile()
+            interactive = False
+            username = None
+            password = None
+            service_principal = None
+            tenant = None
+            if method_index == 0: # device auth
+                interactive = True
+            elif method_index == 1: # username and password
+                username = input('Username: ')
+                password = getpass.getpass('Password: ')
+            elif method_index == 2: # service principal with secret
+                service_principal = True
+                username = input('Service principal: ')
+                tenant = input('Tenant: ')
+                password = getpass.getpass('Client secret: ')
+            elif method_index == 3: # skip
+                return
+            try:
+                profile.find_subscriptions_on_login(
+                    interactive,
+                    username,
+                    password,
+                    service_principal,
+                    tenant)
+                login_successful = True
+                logger.warning('Login successful!')
+            except AdalError as err:
+                logger.error('Login error!')
+                logger.error(err)
+
+def _create_or_update_env(env_name=None):
+    if not env_name:
+        # TODO Support new env creation
+        print('This feature is coming soon.\n')
+        sys.exit()
+    # get the config parser for this env
+    env_config = configparser.SafeConfigParser()
+    env_config.read(os.path.join(ENV_CONFIG_DIR, env_name))
+    # prompt user to choose cloud for env
+    selected_cloud_index = prompt_choice_list(MSG_PROMPT_WHICH_CLOUD, CLOUD_LIST,
+                                              default=get_default_from_config(env_config,
+                                                                              'cloud',
+                                                                              'name',
+                                                                              CLOUD_LIST))
+    if CLOUD_LIST[selected_cloud_index]['name'] != 'public-azure':
+        # TODO support other clouds
+        print('Support for other clouds is coming soon.\n')
+        sys.exit()
+    try:
+        env_config.add_section('cloud')
+    except configparser.DuplicateSectionError:
+        pass
+    env_config.set('cloud', 'name', CLOUD_LIST[selected_cloud_index]['name'])
```

Comment:
The env name and cloud name are different.
e.g.
I can have an env name called 'derek-env1' and this can use cloud 'public-azure'.
So the config file for the 'derek-env1' environment would look like this:
Filename: `.../env_config/derek-env1`
File contents:
```
[cloud]
name = public-azure
```